File: src/main/java/ru/ioleksiv/telegram/bot/api/model/method/game/GetGameHighScores.java
package ru.ioleksiv.telegram.bot.api.model.method.game;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.api.model.NetworkError;
import ru.ioleksiv.telegram.bot.api.model.objects.std.game.GameHighScore;
import ru.ioleksiv.telegram.bot.core.model.method.ChatAction;
import ru.ioleksiv.telegram.bot.core.model.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.responses.ResponseCollection;
import ru.ioleksiv.telegram.bot.core.controller.network.Networker;

import java.util.List;

/**
 * @see <a href="https://core.telegram.org/bots/api#getgamehighscores>getGameHighScores</a>
 */
public class GetGameHighScores extends ChatAction<List<GameHighScore>> {
    private static final String METHOD = "getGameHighScores";

    /**
     * user_id	Integer	Yes	Target user id
     */
    @JsonProperty("user_id")
    private Long userId = null;
    /**
     * message_id	Integer	Optional	Required if inline_message_id is not specified. Identifier of the sent message
     */
    @JsonProperty("message_id")
    private Long messageId = null;
    /**
     * inline_message_id	String	Optional	Required if chat_id and message_id are not specified. Identifier of the inline message
     */
    @JsonProperty("inline_message_id")
    private String inlineMessageId = null;

    public GetGameHighScores(Networker networker) {
        super(METHOD, networker);
    }

    public Long getUserId() {
        return userId;
    }

    public GetGameHighScores setUserId(Long userId) {
        this.userId = userId;
        return this;
    }

    public Long getMessageId() {
        return messageId;
    }

    public GetGameHighScores setMessageId(Long messageId) {
        this.messageId = messageId;
        return this;
    }

    public String getInlineMessageId() {
        return inlineMessageId;
    }

    public GetGameHighScores setInlineMessageId(String inlineMessageId) {
        this.inlineMessageId = inlineMessageId;
        return this;
    }

    @Override
    protected Class<? extends CommonResponse<List<GameHighScore>>> getResultWrapperClass() {
        return ResponseCollection.GameHightScoresResponse.class;
    }

    @Override
    public GetGameHighScores setChatId(Long chatId) {
        pSetChatId(chatId);
        return this;
    }

    @Override
    public GetGameHighScores setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/game/GetGameHighScores.kt
package io.github.oleksivio.tl.kbot.server.api.method.game

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.GameHighScoresResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.common.ChatAction
import io.github.oleksivio.tl.kbot.server.api.objects.ChatId
import io.github.oleksivio.tl.kbot.server.api.objects.std.game.GameHighScore

/**
 * @see [getGameHighScores](https://core.telegram.org/bots/api/#getgamehighscores)
 */
data class GetGameHighScores(
    /**
     * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel
     */
    @JsonProperty(ApiDict.CHAT_ID_KEY)
    override val chatId: ChatId,
    /**
     * user_id Integer Yes Target user id
     */
    @JsonProperty(ApiDict.USER_ID_KEY)
    val userId: Long,
    /**
     * message_id [Integer] OptionalRequired if inline_message_id is not specified. Identifier of the sent message
     */
    @JsonProperty(ApiDict.MESSAGE_ID_KEY)
    val messageId: Long? = null,
    /**
     * inline_message_id [String] Optional Required if chat_id and message_id are not specified. Identifier of the inline message
     */
    @JsonProperty(ApiDict.INLINE_MESSAGE_ID_KEY)
    val inlineMessageId: String? = null
) : ChatAction<List<GameHighScore>>() {

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "getGameHighScores"

    @JsonIgnore
    override val resultWrapperClass = GameHighScoresResponse::class
}
